### PR TITLE
Improve migration docs

### PR DESCRIPTION
**Pull request by Peter-Darton-i2**
The existing v11 -> v12 migration documentation is missing some necessary information.
The net effect is that, if one follows the documented migration process as-is, it doesn't work.

- step 2 needs to pack before it can publish
- the migration tool doesn't adjust the orb-publish build instructions within config.yml and test-deploy.yml so this needs to be documented instead
- step2 parameters were wrong (see also #216 which also contains a fix for this)

This PR fixes that.
... and also makes this orb's test-deploy (step2) build "practise what it preaches" by avoiding the dev-publish step for forked-PR builds (as they should fail to get a context).

### CODE REVIEW

**Comment by KyleTryon** [comment]
A little late here but if you are willing to rebase, we have got the examples update so we can safely remove them from this PR. Thanks for the help!

**Comment by Peter-Darton-i2** [comment]
I've rebased the PR as requested.

Note: You can _safely_ enable the CircleCI -> Project settings -> Advanced "Build forked PRs" flag for this PR ... and once this is merged, you can safely leave it enabled because this PR turns off the dev-publish job (which requires a context) for forked PRs.
i.e. once this is "in" you can let CircleCI do the builds that GitHub has been told to insist upon.

(and if you update other orbs' test-deploy.yml files to follow this pattern, you can then enable forked-PR builds for those too, which will really help non-CircleCI folks contribute usefully to the orb code and generally make your job a little easier 😉)

PS. ensure that the "Pass secrets to builds from forked pull requests" flag is OFF (it defaults to "on"). That setting should never be "on" for a publicly-visible repo.

**Comment by KyleTryon** [comment]
Highly appreciate the help. Unfortunately it was not (only) the publish job that forces us to disable forked builds. We may still experiment with turning it back on, but the original reason was due to crypto miners. We used to get a LOT of spam PRs that totally changed the config to mine crypto. We actually have a config policy feature that I haven't yet to dove too deeply into which can probably help too.

**Comment by Peter-Darton-i2** [comment]
Crypto miners? Wow - that's "unsociable" ... but I guess it's not entirely unexpected; any loophole will be exploited if it can be.

I've been told that OPA config policies are "the CircleCI answer" to all of those security concerns ... although I've personally found that OPA syntax rather difficult; if you do find an answer then please do share because we want to have some of our own CircleCI builds be public and forkable too...

File: .circleci/test-deploy.yml
Comment (KyleTryon) [inline .circleci/test-deploy.yml:1]: Please remove all changes to the CI process to keep this a docs only change. Was there a reason for these config changes?

**Review by KyleTryon — CHANGES_REQUESTED** [review]

File: .circleci/test-deploy.yml
Comment (Peter-Darton-i2) [reply .circleci/test-deploy.yml:1]: The reasons are two-fold.

## Improved example of how to build an orb

This orb is the orb that builds orbs; it is "the primary source of truth" when it comes to building orbs.
As a result, the documentation here (especially the example code) in this orb contains the most "official" recommendation for how to build an orb.

This PR updates the example orb-build code such that CI builds from forked-PRs _don't_ try to do a dev-publish (as they wouldn't have access to the context, e.g. as demonstrated by the failure in [this orb PR](https://github.com/CircleCI-Public/BATS-orb/pull/6)).
i.e. this makes the orb build process suitable for use "in the more general usecase".

This change is an essential step on the road towards enabling forked-PR builds on CircleCI's public orbs.
While I accept that this step isn't the _only_ thing that'll need to change (as per your previous comments about malicious PRs), it is the only thing that will need to change within _this_ orb's source code - all the other concerns will be solved by changes in other places.

i.e. this change to "how we should all be building orbs" is "the final link in the chain" towards having viable public CI builds of forked-PRs on orb code.


## Dogfooding

aka ["Eating your own dog food"](https://en.wikipedia.org/wiki/Eating_your_own_dog_food) aka "Drinking our own champagne".
If we're going to tell everyone else to do this, we should "Put our money where our mouth is" and do it ourselves.

... and because this orb is, itself, an orb, it should follow its own recommendations (otherwise we'll look insincere and hypocritical).


## In summary

We should all be doing orb builds in a forked-PR-friendly manner, including CircleCI's orb-tools orb.

File: src/examples/step2_test-deploy.yml [hunk 1]
```diff
@@ -34,13 +34,30 @@ usage:
...
                 only: /.*/
         # Because our publishing job has a tag filter, we must also apply a filter to each job it depends on.
+        - orb-tools/pack:
+            requires:
+              - command-tests
+              - <my-orb>/my_job
+            filters:
+              tags:
+                only: /.*/
+        - orb-tools/publish:
```
Comment (KyleTryon) [inline src/examples/step2_test-deploy.yml:44]: I am not sure we need to bake in the dev orb publishing process by default. It may be good for documentation but I'm worried it could also complicate the example as it is not a required step.

File: .circleci/test-deploy.yml
Comment (KyleTryon) [reply .circleci/test-deploy.yml:1]: While I agree with the changes shown here should be added, we can not accept them as part of the same PR that makes changes to the docs, as we are going to be required to squash the commits and rename them using our conventional commit standard.

I recommend we break this up into at least two separate PRs, one for the migration guide, and one for the CI and usage example changes.

**Review by KyleTryon — CHANGES_REQUESTED** [review]

**Comment by Peter-Darton-i2** [comment]
OK, understood.
I'll split things up ... although y'all have to merge "both together" or "neither" in order to avoid inconsistencies. As is, this PR keeps everything nice and consistent, ensuring that the orb itself builds using code that's really similar to the example code.

(Sorry for the delay in responding; had an end-of-year work-rush, followed by xmas break. I'll get on with this as soon as I've cleared the holiday backlog)

**Comment by Peter-Darton-i2** [comment]
I've split this PR up. See #229 and #230.

* This one (#220) just updates the migration documentation ... although the new documentation does point people at the example & build code from this orb which are updated in #229 and #230.
* #229 updates the example.
* #230 updates this orb's build code so that it builds the way it's telling everyone else to do it.

**Comment by david-montano-circleci** [comment]
Hello @Peter-Darton-i2 , I am sorry this took so long to review.
Would it be possible for you to update your three PRs so I can review making sure these PRs are based from the latest changes of master?
Thank you!
